lib/websocket-endpoint.js

```diff
@@ -147,7 +147,8 @@ class WebsocketEndpoint {
     drain() {
         var self = this;
 
-        if (self._ws.readyState !== WebSocket.OPEN) {
+        if (self._ws.readyState !== WebSocket.OPEN ||
+            self._message_queue.isEmpty()) {
```

Comment:
Ah, I see. I think using underscores to convey privateness is a fine way to go. Or a comment above the function that it should be private.